### PR TITLE
JhiResolvePagingParams 

### PR DESCRIPTION
**Pull request by azizkhani**
add JhiResolvePagingParams and test for that to move all resolver for paging to ng-jhipster

### CODE REVIEW

**Comment by deepu105** [comment]
@azizkhani I still haven't found time to take a look at the issue you are facing, I can't promise when I'll find time as i'll be busy with Devoxx FR, vacation etc. But since this is not urgent I guess it can wait

**Comment by azizkhani** [comment]
Ok .I starting change generator and waiting for view that  👍

File: src/config.ts [hunk 1]
```diff
@@ -20,17 +20,17 @@ import { Injectable } from '@angular/core';
...
-    i18nEnabled? = false;
-    defaultI18nLang? = 'en';
-    noi18nMessage? = 'translation-not-found';
-    alertAsToast? = false;
-    alertTimeout? = 5000;
-    classBadgeTrue? = 'badge badge-success';
-    classBadgeFalse? = 'badge badge-danger';
-    classTrue? = 'fa fa-lg fa-check text-success';
-    classFalse? = 'fa fa-lg fa-times text-danger';
+    sortIcon ?= 'fa-sort';
```
Comment (deepu105) [inline src/config.ts]: please revert this file change its unrelated

File: src/service/resolve-paging-params.service.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+import { Injectable } from '@angular/core';
+import { Resolve, ActivatedRouteSnapshot, RouterStateSnapshot } from '@angular/router';
+import { JhiPaginationUtil } from '.';
```
Comment (deepu105) [inline src/service/resolve-paging-params.service.ts]: import directly the file as its in same folder to avoid circular ref

File: yarn-error.log [hunk 3]
```diff
@@ -0,0 +1,4040 @@
+Arguments:
```
Comment (deepu105) [inline yarn-error.log]: revert this file, and add it to gitignore

File: src/service/resolve-paging-params.service.ts (hunk 2 above)
Comment (azizkhani) [reply src/service/resolve-paging-params.service.ts]: import {JhiPaginationUtil} from "./pagination-util.service"; ???

Comment (deepu105) [reply src/service/resolve-paging-params.service.ts]: yes

File: package.json [hunk 4]
```diff
@@ -32,6 +32,7 @@
     "@angular/compiler": "^5.0.0",
     "@angular/compiler-cli": "^5.0.0",
     "@angular/core": "^5.0.0",
+    "@angular/router": "^5.0.0",
```
[the remark this answers is not in the document]
Comment (deepu105) [reply package.json]: you need to add it to both peer dependency and dev dependency

File: package.json [hunk 5]
```diff
@@ -56,7 +57,7 @@
     "loader-utils": "^1.1.0",
     "reflect-metadata": "^0.1.10",
     "rimraf": "^2.6.1",
-    "rxjs": "^5.5.5",
+    "rxjs": "^5.5.6",
```
[the remark this answers is not in the document]
Comment (deepu105) [reply package.json]: its ok

File: src/service/resolve-paging-params.service.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+import { Injectable } from '@angular/core';
+import { Resolve, ActivatedRouteSnapshot, RouterStateSnapshot } from '@angular/router';
+import {JhiPaginationUtil} from './pagination-util.service';
```
Comment (deepu105) [inline src/service/resolve-paging-params.service.ts]: lint is failing here, run `yarn lint:fix`

Comment (azizkhani) [reply src/service/resolve-paging-params.service.ts]: when i run yarn lint:fix ,fixed config.ts instead off resolve-paging-params.service.ts

**Comment by azizkhani** [comment]
@deepu105 can you help me to resolve this test?

**Comment by azizkhani** [comment]
@deepu105 i changed test and pass all test. 👍

**Pull request merged**